core/chains/evm/gas/rollups/op_l1_oracle.go

```diff
@@ -443,6 +463,33 @@ func (o *optimismL1Oracle) getV1GasPrice(ctx context.Context) (*big.Int, error)
 	return new(big.Int).SetBytes(b), nil
 }
 
+// Returns the gas price for Mantle. The formula is the same as Optimism Bedrock (getV1GasPrice), but the tokenRatio parameter is multiplied
+func (o *optimismL1Oracle) getMantleGasPrice(ctx context.Context) (*big.Int, error) {
+	// set oracle address
+	l1OracleAddress := common.HexToAddress(o.l1OracleAddress)
+
+	// call oracle to get tokenRatio
+	b, err := o.client.CallContract(ctx, ethereum.CallMsg{
+		To:   &l1OracleAddress,
+		Data: o.tokenRatioCalldata,
+	}, nil)
+	if err != nil {
+		return nil, fmt.Errorf("tokenRatio() call failed: %w", err)
+	}
+
+	// create bigInt from returned bytes
+	tokenRatio := new(big.Int).SetBytes(b)
+
+	// call optimism bedrock gas price function
+	v1GasPrice, err := o.getV1GasPrice(ctx)
```

Comment:
Just realized that the OP V1 gas price method is only really calling `l1BaseFee()`. Could we potentially just do a batch call here for both `tokenRatio()` and `l1BaseFee90` so we can get the results in just one RPC call? The `getEcotoneFjordGasPrice` method should have an example of how that would look.